src/dune/lib_name.ml

```diff
@@ -20,13 +28,20 @@ module Local = struct
 
       let description_of_valid_string = Some valid_format_doc
 
-      let valid_char = function
-        | 'A' .. 'Z'
-        | 'a' .. 'z'
-        | '_'
-        | '0' .. '9' ->
-          true
-        | _ -> false
+      let hint_valid =
+        Some
+          (fun name ->
+            String.to_seq name
+            |> Seq.filter_map ~f:(fun c ->
+                   if not (valid_char c) then
```

Comment:
Could you invert the clauses of the `if` and remove the `not`? It will make it simpler to read.